File: ts-sdk/src/sdk/models/operations/readscope.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class ReadScopePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=scopeId" })
  scopeId: string;
}

export class ReadScopeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ReadScopePathParams;
}

export class ReadScopeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  readScopeResponse?: shared.ReadScopeResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/readscope.go
package operations

import (
	"github.com/speakeasy-sdks/formance-sdks/go-client-sdk/pkg/models/shared"
)

type ReadScopePathParams struct {
	ScopeID string `pathParam:"style=simple,explode=false,name=scopeId"`
}

type ReadScopeRequest struct {
	PathParams ReadScopePathParams
}

type ReadScopeResponse struct {
	ContentType       string
	ReadScopeResponse *shared.ReadScopeResponse
	StatusCode        int
}
